solution/ui/regulations/eregs-vite/src/views/PolicyRepository.vue

```diff
@@ -86,8 +106,176 @@ const getDocList = async () => {
     }
 };
 
-getDocList();
+// use "reactive" method to make urlParams reactive when provided/injected
+// selectedParams.paramString is used as the reactive prop
+const selectedParams = reactive({
+    paramString: "",
+    paramsArray: [],
+});
+
+// method to add selected params
+const addSelectedParams = (paramArgs) => {
+    const { id, name, type } = paramArgs;
+
+    // early return if the param is already selected
+    if (selectedParams.paramString.includes(`${type}=${id}`)) return;
+
+    // update paramString that is used as reactive prop for watch
+    if (selectedParams.paramString) {
+        selectedParams.paramString += `&${type}=${id}`;
+    } else {
+        selectedParams.paramString = `?${type}=${id}`;
+    }
+
+    // create new selectedParams key for array of objects for selections
+    selectedParams.paramsArray.push({ id, name, type });
+};
+
+// method to remove selected params
+const removeSelectedParams = (paramArgs) => {
+    const { id, type } = paramArgs;
+
+    // update paramString that is used as reactive prop for watch
+    const paramStringArray = selectedParams.paramString.split("&");
+    const paramString = paramStringArray
+        .filter((param) => !param.includes(`${type}=${id}`))
+        .join("&");
+
+    selectedParams.paramString = paramString;
+
+    // create new selectedParams key for array of objects for selections
+    selectedParams.paramsArray = selectedParams.paramsArray.filter(
+        (param) => param.id != id
+    );
+};
+
+const clearSelectedParams = () => {
+    selectedParams.paramString = "";
+    selectedParams.paramsArray = [];
+};
+
+provide("selectedParams", selectedParams);
+
+// utility method to parse $route.query to return `${key}=${value},${value}` string
+const getRequestParams = (query) => {
+    const requestParams = Object.entries(query)
+        .map(([key, value]) => {
+            const valueArray = value
+                .split(",")
+                .filter((id) => !Number.isNaN(parseInt(id, 10)));
+            return valueArray.map((v) => `${key}=${v}`).join("&");
+        })
+        .join("&");
+
+    return requestParams;
+};
+
+// policyDocSubjects fetch for subject selector
+// fetch here so we have it in context; pass down to selector via props
+const policyDocSubjects = ref({
+    results: [],
+    loading: true,
+});
+
+// called on load
+const getDocSubjects = async () => {
+    try {
+        const subjectsResponse = await getPolicyDocSubjects({
+            apiUrl: props.apiUrl,
+        });
+
+        policyDocSubjects.value.results = subjectsResponse.sort(sortSubjects);
```

Comment:
This is setting this in alpha order?  Why not just have this done on the backend and remove that extra processing?

Reply:
Actually might be better to do this on the backend anyways.  We have a "true sort" method on the backend we can use that would allow us to take things like numbers into consideration when it comes to sorting things.  We can easily break this ticket out but JS might be looking at this as a straight string which can cause weird ordering if there are numbers in subject names(if this is a thing)

Reply:
Sorting in the browser isn't inefficient (see [perf stats here](https://www.measurethat.net/Benchmarks/ShowResult/465724)) but we could move sorting to the back end if we wanted to. 

Right now, though, there's still a question of which name will be used (`full_name`, `short_name`, `abbreviation`).  Keeping it on the front end lets us easily change that hierarchy; once its solidified we could move it to the back end if needed.